iceoryx_hoofs/source/posix_wrapper/timer.cpp

```diff
@@ -113,11 +112,7 @@ void Timer::OsTimer::callbackHelper(sigval data) noexcept
             // prohibits other threads from entering the flag protected region
             handle.m_callbackIsAboutToBeExecuted.test_and_set(std::memory_order_acq_rel);
 
-            if (handle.m_timer == nullptr)
-            {
-                errorHandler(Error::kPOSIX_TIMER__INCONSISTENT_STATE);
-                return;
-            }
+            cxx::Expects((handle.m_timer != nullptr) && "Timer in inconsistent state");
```

Comment:
In a future version of `Expects` the message should be a separate (optional?) argument and not appended with `&&` like in the good old days for `assert` in C.

Only a style issue and personal opinion though, nothing for this PR unless this is already possible.